src/test/scala/cr/pulsar/PulsarSpec.scala

```diff
@@ -16,56 +16,28 @@
 
 package cr.pulsar
 
-import cats.{ Eq, Inject }
 import cats.effect._
-import cats.effect.concurrent.Deferred
+import cats.effect.concurrent.{ Deferred, Ref }
 import cats.implicits._
-import cr.pulsar.Config._
+import cr.pulsar.Publisher.MessageKey
 import fs2.Stream
-import munit.FunSuite
 import org.apache.pulsar.client.api.SubscriptionInitialPosition
-import scala.concurrent.ExecutionContext
 
-class PulsarSpec extends FunSuite {
+class PulsarSpec extends PulsarSuite {
 
-  implicit val `⏳` = IO.contextShift(ExecutionContext.global)
-  implicit val `⏰` = IO.timer(ExecutionContext.global)
-
-  override def munitValueTransforms: List[ValueTransform] =
-    super.munitValueTransforms :+ new ValueTransform("IO", {
-          case ioa: IO[_] => IO.suspend(ioa).unsafeToFuture
-        })
-
-  case class Event(value: String)
-
-  object Event {
-    implicit val eq: Eq[Event] = Eq.fromUniversalEquals
-
-    implicit val inject: Inject[Event, Array[Byte]] =
-      new Inject[Event, Array[Byte]] {
-        def inj: Event => Array[Byte]         = _.value.getBytes("UTF-8")
-        def prj: Array[Byte] => Option[Event] = bs => Event(new String(bs, "UTF-8")).some
-      }
-  }
-
-  val cfg = Config(
-    PulsarTenant("public"),
-    PulsarNamespace("default"),
-    PulsarURL("pulsar://localhost:6650")
-  )
-  val subs   = Subscription(Subscription.Name("test"), Subscription.Type.Failover)
-  val spos   = SubscriptionInitialPosition.Latest
-  val topic  = Topic(cfg, Topic.Name("test"), Topic.Type.Persistent)
-  val batch  = Publisher.Batching.Disabled
-  val msgKey = Publisher.MessageKey.Default
+  val subs  = Subscription(Subscription.Name("test"), Subscription.Type.Failover)
+  val spos  = SubscriptionInitialPosition.Latest
+  val topic = Topic(cfg, Topic.Name("test"), Topic.Type.Persistent)
+  val batch = Publisher.Batching.Disabled
+  val shard = (_: Event) => Publisher.MessageKey.Default
 
   test("A message is published and consumed successfully") {
     val res: Resource[IO, (Consumer[IO], Publisher[IO, Event])] =
       for {
         client <- PulsarClient.create[IO](cfg.serviceUrl)
         consumer <- Consumer.create[IO](client, topic, subs, spos)
         blocker <- Blocker[IO]
-        publisher <- Publisher.create[IO, Event](client, topic, msgKey, batch, blocker)
```

Comment:
Are we going to unify the naming to `Producer`?

Reply:
That should be a different PR